content/dev/blog/tidyeval.Rmd

```diff
@@ -0,0 +1,232 @@
+        geom_point() +
+        theme_minimal()
+}
+
+plot_2d <- function(df, vars){
+    groups <- vars[vars %in% groups(df)]
+    measure <- vars[length(vars)][[1]]
+    df %>%
+        select(!!!groups, !!measure) %>%
+        arrange(desc(!!measure)) %>%
+        ggplot(aes(x = !!measure, y = !!groups[[1]], , color = !!groups[[2]])) +
+        geom_point() +
+        theme_minimal()
+}
+```
+
+At Crunch, one of the ways we try to make data exploration simple is by providing sensible default views that take into account the properties of your data and metadata. We've just released some plotting methods in our `crplyr` R package that define methods for `ggplot2`'s [`autoplot()`](https://ggplot2.tidyverse.org/reference/autoplot.html) function.
+`autoplot()` was the ideal approach for us to encapsulate the logic of how to "just do the right thing." You can do the analysis you want, and it will make smart choices about how to display it with no additional input--all of which you can control or override with additional `ggplot2` layers, if you want.
+
+This lets us do [insert examples here, with pictures]
+
+Making it look easy, though, can be hard work. Our `autoplot` methods inspect the input objects to understand their dimensionality and data types and choose an appropriate visualization. Figuring out and passing the right arguments to the right places can be messy, so make these functions work, we took advantage of "[tidyeval](https://dplyr.tidyverse.org/articles/programming.html)", a new framework that systematizes non-standard evaluation in R.
+
+To illustrate this pattern, this blog post goes through a simplified example using the "diamonds" example dataset and pure `dplyr` methods. We'll create a plotting function that adjusts to the number of grouping variables in a tibble. This lets you pipe data from a dplyr pipeline into a single function and get a meaningful, appropriate plot, like this:
+
+```{r}
+library(ggplot2)
+library(dplyr)
+diamonds %>%
+    group_by(cut, clarity) %>%
+    tally() %>%
+    autoplot()
+
+diamonds %>%
+    group_by(clarity, cut, color) %>%
+    summarize(mean_price = mean(price)) %>%
+    autoplot()
+```
+
+The actual code for our Crunch `autoplot` methods is [here](https://github.com/Crunch-io/crplyr/blob/ggplot-methods/R/plotting.R).
+
+## Non-Standard Evaluation: Great Power, Great Pain
+
+A great feature the R language is that it lets you access and manipulate the environment in which a function is called. This non-standard evaluation (NSE) gives package authors a flexible and powerful way to build programming interfaces.
+
+At Crunch, we use NSE a lot. One example is in reporting better error messages. If you send the wrong data to the API, you might get back a response like `400: Payload is malformed`. That's not very helpful even for users who know our API well. We use NSE to inspect the user's calling environment, figure out which variables or data structures are causing the problem, and suggest a fix.
```

Comment:
Perhaps include a link to an example in `crunch` and show what a better error message looks like.